server/github.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"github.com/andygrunwald/go-jira"
+	"github.com/google/go-github/v25/github"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+func httpGithubEvent(ji Instance, w http.ResponseWriter, r *http.Request) (int, error) {
+	mattermostUserId := r.Header.Get("Mattermost-User-Id")
+	if mattermostUserId == "" {
+		return http.StatusUnauthorized, errors.New("Not authorized")
+	}
+
+	jiraUser, err := ji.GetPlugin().userStore.LoadJIRAUser(ji, mattermostUserId)
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	client, err := ji.GetClient(jiraUser)
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return http.StatusBadRequest, err
+	}
+
+	event, _ := github.ParseWebHook(github.WebHookType(r), body)
+	plugin := ji.GetPlugin()
+
+	switch event := event.(type) {
+	case *github.PullRequestEvent:
+		plugin.askAssigneeToClosePRRelatedIssues(client, event)
+	}
+
+	return http.StatusOK, nil
+}
+
+func (p *Plugin) askAssigneeToClosePRRelatedIssues(client Client, event *github.PullRequestEvent) error {
```

Comment:
I think the PR submitter is more appropriate to receive this message because they have full knowledge of what was merged, and so can provide the best context for the further testing of the PR while resolving. Some fields required to resolve would likely need to be entered by the PR submitter.

We should still confirm that the user is connected to the Jira plugin, however.

Reply:
Ok! Then maybe we can include a custom post type with several buttons (one for each possible transition of the issue) and let the user (PR submitter) select the new transition for the ticket

Reply:
For the first iteration of this feature, I think we should just link the user to the ticket and let them handle it from there. Some transitions require more fields to be filled out, and without functionality to fill those out, the feature will not function properly.

We can get the fields using functionality discussed here: https://community.atlassian.com/t5/Jira-questions/Get-Required-mandatory-transition-fields-using-REST-API/qaq-p/594308 . But I think it is out of scope for the first iteration of this feature. There's already a decent amount of design additions here.